c_src/eleveldb.cc

```diff
@@ -705,6 +714,7 @@ async_write(
     ERL_NIF_TERM result = fold(env, argv[2], write_batch_item, *batch);
     if(eleveldb::ATOM_OK != result)
     {
+        delete batch;
```

Comment:
Would be good to add a comment (perhaps above where the batch variable is defined) indicating that the lifetime of the batch variable is managed by the WriteTask, created below. We only need to delete it here since we haven't yet handed it to the WriteTask object.

Even better (or at least more modern C++) would be to make batch an auto_ptr and call auto_ptr<>.release() in the WriteTask constructor. That way we make clear who owns the WriteBatch object at any given moment.